File: Funciones/17. Factorial_while/js/factorial.js
/**
 * Factorial de 5
 * Autor: Santiago GM
 * Fecha: 03/04/2024
 */

let numero;

//como paramétro

function factor(pnumero){
    numero = pnumero;
    let factorial;
    let contador;

    contador = 0;
    factorial = 1;

    while(contador < numero){
        contador = contador+1;
        factorial =factorial*contador;
    } 
    return factorial;
}

//Como Expresión

const factorExp = function (pnumero){
    numero = pnumero;
    let factorial;
    let contador;

    contador = 0;
    factorial = 1;

    while(contador < numero){
        contador = contador+1;
        factorial =factorial*contador;
    } 
    return factorial;
}

File: python/21. factorialFor/utils/factorial.py
#Factorial de 5 con for
#Autor: Santiago GM

#Como paramétro

def factorial(number):
    factorial = 1
    contador = 1
    for contador in range(contador, number + 1):
        factorial *= contador
    return factorial

#Función lambda

factorial_lamb = lambda  number: 1 if number == 0 else number * factorial_lamb(number - 1)
